src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactory.cs

```diff
@@ -675,9 +677,18 @@ private bool TryCompileInMemoryAssembly(IBuildEngine buildEngine, RoslynCodeTask
             }
 
             // The source code cannot actually be compiled "in memory" so instead the source code is written to disk in
-            // the temp folder as well as the assembly.  After compilation, the source code and assembly are deleted.
+            // the temp folder as well as the assembly. After build, the source code and assembly are deleted.
             string sourceCodePath = FileUtilities.GetTemporaryFileName(".tmp");
-            string assemblyPath = FileUtilities.GetTemporaryFileName(".dll");
+
+            string processSpecificInlineTaskDir = Path.Combine(
```

Comment:
Not sure I get how does this path propagates further to be used out-of-proc?

Reply:
good question! Since I changed the loading to do 
```csharp
assembly = Assembly.LoadFrom(assemblyPath);
```
instead of 
```csharp
assembly = Assembly.Load(File.ReadAllBytes(assemblyPath));
```
reflect the task from the assembly and the task points to its assembly with valid `Location` property.

then in `TaskExecutionHost` in the code I added it's wrapped in `TaskHostTask` and this already works such that that the location guides how to load it in a TaskHost node (this mechanism works already for "non-custom-Factory-tasks" so getting the location was the only obstacle).